cmk/base/plugins/agent_based/cisco_asa_sensors.py

```diff
@@ -0,0 +1,340 @@
+
+
+def check_cisco_asa_temp(item, params: TempParamType, section) -> CheckResult:
+    try:
+        sensor = section['temp'][item]
+
+        yield Result(state=sensor.status, summary='Status: %s' % sensor.state_readable)
+
+        yield from check_temperature(
+            sensor.value,
+            dev_unit=sensor.unit,
+            dev_status=sensor.status,
+            dev_status_name=sensor.state_readable,
+            params=params,
+            unique_name='check_cisco_asa_temp.%s' % item,
+        )
+    except KeyError:
+        pass
+
+
+register.check_plugin(
+    name='cisco_asa_temp',
+    service_name='Temperature %s',
+    sections=['cisco_asa_sensors'],
+    discovery_function=discovery_cisco_asa_temp,
+    check_function=check_cisco_asa_temp,
+    check_default_parameters={},
+    check_ruleset_name='temperature'
+)
+
+
+# ##################################################################################################
+#
+# ASA SENSORS FAN
+#
+# ##################################################################################################
+
+
+def discovery_cisco_asa_fan(section: Dict) -> DiscoveryResult:
+    for key in section['fan']:
+        yield Service(item=key)
+
+
+def check_cisco_asa_fan(item, params, section) -> CheckResult:
+    try:
+        sensor = section['fan'][item]
+
+        yield Result(state=sensor.status, summary='Status: %s' % sensor.state_readable)
+
+        yield from check_levels(
+            sensor.value,
+            label='Speed',
+            levels_lower=params.get('lower', None),
+            levels_upper=params.get('upper', None),
+            metric_name='fan' if params.get('output_metrics') else None,
+            render_func=lambda v: '%s RPM' % str(v),
+        )
+
+    except KeyError:
+        pass
+
+
+register.check_plugin(
+    name='cisco_asa_fan',
+    service_name='Fan %s',
+    sections=['cisco_asa_sensors'],
+    discovery_function=discovery_cisco_asa_fan,
+    check_function=check_cisco_asa_fan,
+    check_default_parameters={'output_metrics': True},
+    check_ruleset_name='hw_fans'
+)
+
+
+# ##################################################################################################
+#
+# ASA SENSORS POWER SUPPLY
+#
+# ##################################################################################################
+
+def discovery_cisco_asa_power(section: Dict) -> DiscoveryResult:
+    for key in section['power']:
+        yield Service(item=key)
+
+
+def check_cisco_asa_power(item, params, section) -> CheckResult:
+    try:
+        sensor = section['power'][item]
+
+        yield Result(state=sensor.status, summary='Status: %s' % sensor.state_readable)
+
+    except KeyError:
+        pass
+
+
+register.check_plugin(
+    name='cisco_asa_power',
+    service_name='Power %s',
+    sections=['cisco_asa_sensors'],
+    discovery_function=discovery_cisco_asa_power,
+    check_function=check_cisco_asa_power,
+    check_default_parameters={},
```

Comment:
There is no need to use `check_default_parameters` here, since there are no discovery and no check parameters. Please remove this and also remove the `params` argument to `check_cisco_asa_power`.